lib/rules/no-test-file-importing.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @fileOverview no importing of test files are allowed
+ */
+
+'use strict';
+
+const REPORT_MESSAGE
+  = 'Do not import "-test.js" file in a test file. This will cause the module and tests from the imported file to be executed again.';
+
+/**
+ * Returns the name of the module imported or re-exported.
+ *
+ * @param {ASTNode} node - A node to get.
+ * @returns {string} the name of the module, or empty string if no name.
+ */
+function getValue(node) {
+  if (node && node.source && node.source.value) {
+    return node.source.value.trim();
+  }
+
+  return '';
+}
+
+/**
+ * Checks if the name of the import has '-test' in it, and reports if so.
+ *
+ * @param {ASTNode} importDeclaration - The importDeclaration node.
+ * @returns {void} No return value
+ */
+function validateImport(importDeclaration, context) {
+  const importSource = getValue(importDeclaration);
+  const re = /.*-test$/i;
+
+  if (importSource.match(re)) {
```

Comment:
Is there any reason to not just use `endsWith`?

Reply:
Good call, changed to use endsWith.